CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Onboarding/OnboardingViewController.swift

```diff
@@ -8,35 +8,50 @@
 import Foundation
 import UIKit
 
-class OnboardingViewController: UIViewController,
-                                UICollectionViewDelegate,
-                                UICollectionViewDataSource,
-                                UICollectionViewDelegateFlowLayout {
+class OnboardingViewController: UIViewController {
     @IBOutlet weak var pagesCollection: UICollectionView!
     @IBOutlet weak var pageControl: UIPageControl!
     @IBOutlet weak var nextButton: UIButton!
     @IBOutlet weak var containerView: UIStackView!
         @IBOutlet weak var viewContainer: UIStackView!
-    fileprivate var navigationBarBottomAdapter: OnboardingNavigationBarBottomAdapter?
-    let configuration = GiniConfiguration.shared
-    fileprivate func configureCollectionView() {
+    private var navigationBarBottomAdapter: OnboardingNavigationBarBottomAdapter?
+    private (set) var dataSource: OnboardingDataSource
+    private let configuration = GiniConfiguration.shared
+    init() {
+        dataSource = OnboardingDataSource(configuration: configuration)
+        super.init(nibName: "OnboardingViewController", bundle: giniCaptureBundle())
+      }
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    private func configureCollectionView() {
         pagesCollection.register(
             UINib(nibName: "OnboardingPageCell", bundle: giniCaptureBundle()),
-            forCellWithReuseIdentifier: "onboardingPageCellIdentifier")
-        pagesCollection.dataSource = self
-        pagesCollection.delegate = self
+            forCellWithReuseIdentifier: OnboardingPageCell.reuseIdentifier)
         pagesCollection.setNeedsLayout()
         pagesCollection.layoutIfNeeded()
         pagesCollection.reloadData()
+        pagesCollection.isPagingEnabled = true
+        pagesCollection.dataSource = dataSource
+        pagesCollection.delegate = dataSource
+        dataSource.delegate = self
```

Comment:
[Clarification] Why do we need to set 
`dataSource.delegate = self
`?

Reply:
@zladzeyka  We need it to update pageControl based on the displayed cell